.circleci/config.yml

```diff
@@ -0,0 +1,75 @@
+---
+version: 2.0
+
+references:
+  stack_build: &stack_build
+    docker:
+      # https://github.com/haskell-works/stack-build/blob/master/minimal/Dockerfile
+      - image: quay.io/haskell_works/stack-build-minimal
+
+    steps:
+      - checkout
+      - run:
+          name: Digest
+          command: |
+            echo -- "${STACK_ARGUMENTS}" > rdigest
+            git ls-files | xargs md5sum > sdigest
+
+      - restore_cache:
+          keys:
+            - 1-{{ .Branch }}-{{ checksum "rdigest" }}-{{ checksum "sdigest" }}
+            - 1-{{ .Branch }}-{{ checksum "rdigest" }}-
+            - 1-{{ .Branch }}-
+            - 1-master-
```

Comment:
I'm a bit disappointed we lost our inclusion of `stack.yaml` and/or the snapshot in the actual cache key. I love being able to start from a clean slate when the snapshot changes. Not a blocker though.

Reply:
The resolver is included by way of `$STACK_ARGUMENTS`. Were we tracking changes to the contents of `stack.yaml` before? (I'm not familiar with Circle 1.0's syntax exactly)

Reply:
Sure, in the GHC 8.2 case it is. Not for the non GHC 8.2 case since `STACK_ARGUMENTS` doesn't change if the resolver doesn't change.

We were not tracking this in CircleCI 1.0 since you really couldn't with their simple caching system.

For context, the reason we want to totally bust the cache when the snapshot changes is so we don't keep building up new snapshots in `~/.stack` that we don't use any more. That makes caching take a super long time.

Again, not a blocker here. We'll have to think about how to do this. I feel like both GHC 8.2 and the latest GHC need their own `stack.yaml` file so we are consistent, and then we can just checksum the contents of that `stack.yaml` or something.

Reply:
That's what I had in mind as well; I should be able to fit that in here. :)